libs-scala/grpc-utils/src/main/scala/com/digitalasset/grpc/GrpcException.scala

```diff
@@ -3,15 +3,41 @@
 
 package com.digitalasset.grpc
 
+import com.digitalasset.grpc.GrpcStatus.SpecificGrpcStatus
 import io.grpc.{Metadata, Status, StatusException, StatusRuntimeException}
 
 object GrpcException {
-
-  def unapply(arg: Exception): Option[(Status, Metadata)] =
-    arg match {
+  def unapply(exception: Exception): Option[(Status, Metadata)] =
+    exception match {
       case e: StatusRuntimeException => Some((e.getStatus, e.getTrailers))
       case e: StatusException => Some((e.getStatus, e.getTrailers))
       case _ => None
     }
 
+  case class SpecificGrpcException(status: SpecificGrpcStatus) {
```

Comment:
```suggestion
  private[grpc] final class SpecificGrpcException(status: SpecificGrpcStatus) {
```
Since we're implementing `unapply` ourselves I don't think we need to make a `case class` out of this.